api-metastore/src/main/java/org/zalando/nakadi/service/EventTypeService.java

```diff
@@ -521,8 +534,43 @@ private void updateRetentionTime(final EventType original, final EventType event
     private void updateAnnotationsAndLabels(final EventType original, final EventType eventType)
             throws InvalidEventTypeException {
 
+       /* this is a temporary solution to ensure the backward compatibility and it is here because
+        the DataLakeAnnotationValidator.java does not have access to the event type */
+        final List<String> breakingCompatibilitiesNames = Arrays.asList(
+                "content-visibility.inference-log.v3.test",
+                "zoutlets.wm.article-decision",
+                "merchant-price-service.work-log",
+                "merchant-price-service.work-log-raw",
+                "content.customer-targeting.segments.sqate",
+                "merchant-price-service.price-update-called",
+                "merchant-price-service.price-update-processed",
+                "octopus.analysis-metrics-v2",
+                "wholesale.purchase-order-event",
+                "zoutlets.wm.commodity-group.updated",
+                "connected-retail.article-insights-service.configuration-changed",
+                "merchant-price-service.price-threshold-updates",
+                "product-data-quality.inference-pipeline.spp-updates-lineage",
+                "product-data-quality.inference-pipeline.decision-maker-lineage",
+                "product-data-quality.hitl.human-annotation-ui-lineage");
+
         if (eventType.getAnnotations() == null) {
-            eventType.setAnnotations(original.getAnnotations());
+            final Map<String, String> originalAnnotations = original.getAnnotations();
+            if (breakingCompatibilitiesNames.contains(original.getName())){
+                if (!originalAnnotations.containsKey("datalake.zalando.org/retention-period")) {
+                    originalAnnotations.put("datalake.zalando.org/retention-period", "unlimited");
+                }
+                eventType.setAnnotations(originalAnnotations);
```

Comment:
Again, we should not be doing this: the variable is called _original annotations_ for a reason.